Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -356,6 +367,101 @@ lemma Subgraph.IsPerfectMatching.symmDiff_spanningCoe_IsCycles
     use w, w'
     aesop
 
+lemma IsCycles.snd_of_support_of_path_of_adj [Fintype V] [DecidableEq V] {v w w' : V}
+    (hcyc : G.IsCycles) (p : G.Walk v w) (hw : w ≠ w') (hw' : w' ∈ p.support) (hp : p.IsPath)
+    (hadj : G.Adj v w') : p.snd = w' := by
+  apply hp.toSubgraph_adj_sndOfNotNil
+  rw [Walk.mem_support_iff_exists_getVert] at hw'
+  obtain ⟨n, ⟨rfl, hnl⟩⟩ := hw'
+  by_cases hn : n = 0 ∨ n = p.length
+  · aesop
+  have e : G.neighborSet (p.getVert n) ≃ p.toSubgraph.neighborSet (p.getVert n) := by
+    refine @Classical.ofNonempty _ ?_
+    rw [← Cardinal.eq, ← Set.cast_ncard (Set.toFinite _), ← Set.cast_ncard (Set.toFinite _),
+        hp.ncard_neighborSet_toSubgraph_internal_eq_two (by omega) (by omega),
+        hcyc (Set.nonempty_of_mem hadj.symm)]
+  rw [Subgraph.adj_comm, Subgraph.adj_iff_of_neighborSet_equiv e (Set.toFinite _).fintype v]
+  exact hadj.symm
+
+lemma IsCycles.reachable_sdiff_toSubgraph_spanningCoe [Fintype V] [DecidableEq V] {v w : V}
+    (hcyc : G.IsCycles) (p : G.Walk v w) (hp : p.IsPath) :
+    (G \ p.toSubgraph.spanningCoe).Reachable w v := by
+  -- Consider the case when p is nil
+  by_cases hvw : v = w
+  · subst hvw
+    use .nil
+  have hpn : ¬p.Nil := Walk.not_nil_of_ne hvw
+  obtain ⟨w', ⟨⟨hw'1, hw'2⟩, hwu⟩⟩ := hcyc.existsUnique_ne_adj
+    ((Walk.toSubgraph_adj_snd p hpn).adj_sub)
+  -- The edge (v, w) can't be in p, because then it would be the second node
+  have hnpvw' : ¬ p.toSubgraph.Adj v w' := by
+    intro h
+    exact hw'1 (hp.toSubgraph_adj_sndOfNotNil h)
+  -- If w = w', then then the reachability can be proved with just one edge
+  by_cases hww' : w = w'
+  · subst hww'
+    have : (G \ p.toSubgraph.spanningCoe).Adj w v := by
+      simp only [sdiff_adj, Subgraph.spanningCoe_adj]
+      exact ⟨hw'2.symm, fun h ↦ hnpvw' h.symm⟩
+    exact this.reachable
+  -- Construct the walk needed recursively by extending p
+  have hle : (G \ (p.cons hw'2.symm).toSubgraph.spanningCoe) ≤ (G \ p.toSubgraph.spanningCoe) := by
+    apply sdiff_le_sdiff (by rfl) ?hcd
+    aesop
+  have hp'p : (p.cons hw'2.symm).IsPath := by
+    rw [Walk.cons_isPath_iff]
+    refine ⟨hp, fun hw' ↦ ?_⟩
+    exact hw'1 (hcyc.snd_of_support_of_path_of_adj _ hww' hw' hp hw'2)
+  have : (G \ p.toSubgraph.spanningCoe).Adj w' v := by
+    simp only [sdiff_adj, Subgraph.spanningCoe_adj]
+    refine ⟨hw'2.symm, fun h ↦ ?_⟩
+    exact hnpvw' h.symm
+  use (((hcyc.reachable_sdiff_toSubgraph_spanningCoe
+    (p.cons hw'2.symm) hp'p).some).mapLe hle).append this.toWalk
+termination_by Fintype.card V + 1 - p.length
+decreasing_by
+  simp_wf
+  have := Walk.IsPath.length_lt hp
+  omega
+
+lemma IsCycles.reachable_deleteEdge [Fintype V] [DecidableEq V] (hadj : G.Adj v w)
+    (hcyc : G.IsCycles) : (G \ SimpleGraph.fromEdgeSet {s(v, w)}).Reachable v w := by
```

Comment:
```suggestion
lemma IsCycles.reachable_deleteEdges [Fintype V] [DecidableEq V] (hadj : G.Adj v w)
    (hcyc : G.IsCycles) : (G.deleteEdges {s(v, w)}).Reachable v w := by
```
This would justify the name